src/main/java/com/jwplayer/southpaw/state/RocksDBState.java

```diff
@@ -243,39 +252,20 @@ public RocksDBState(Map<String, Object> config) {
     public void backup() {
         logger.info("Backing up RocksDB state");
         try {
-            switch(backupURI.getScheme().toLowerCase()) {
-                case FileHelper.SCHEME:
-                    backup(backupURI.getPath());
-                    break;
-                case S3Helper.SCHEME:
-                    String localBackupPath = getLocalBackupPath(uri);
-                    backup(localBackupPath);
-                    s3Helper.syncToS3(new URI(localBackupPath), backupURI);
-                    break;
-                default:
-                    throw new RuntimeException("Unsupported schema: " + backupURI.getScheme());
-            }
-        } catch(InterruptedException | ExecutionException | URISyntaxException | RocksDBException ex) {
+            backupEngine.createNewBackup(rocksDB, true);
+            backupEngine.purgeOldBackups(backupsToKeep);
+        } catch(RocksDBException ex) {
             throw new RuntimeException(ex);
         }
-        logger.info("RocksDB state backup complete");
-    }
 
-    /**
-     * Backups the DB to a local path.
-     * @param backupPath - The local backup path
-     */
-    protected void backup(String backupPath) throws RocksDBException {
-        File file = new File(backupPath);
-        if(!file.exists()) file.mkdir();
-        logger.info("Opening RocksDB backup engine");
-        try (final BackupableDBOptions backupOptions = new BackupableDBOptions(backupPath)
-                .setShareTableFiles(true)
-                .setMaxBackgroundOperations(parallelism);
-            final BackupEngine backupEngine = BackupEngine.open(Env.getDefault(), backupOptions)) {
-            backupEngine.createNewBackup(rocksDB, true);
-            backupEngine.purgeOldBackups(backupsToKeep);
+        if(s3Helper != null) {
+            try {
+                s3Helper.syncToS3(new URI(backupPath), backupURI);
+            } catch(InterruptedException | ExecutionException | URISyntaxException ex) {
+                throw new RuntimeException(ex);
+            }
         }
+        logger.info("RocksDB state backup complete");
```

Comment:
add within finally?

Reply:
I'd think if this excepts out we wouldn't want to print this log line